exporter/kafkaexporter/kafka_exporter_test.go

```diff
@@ -912,75 +1252,83 @@ func newMockLogsExporter(t *testing.T, cfg Config, host component.Host) (*kafkaE
-		}
+func newKgoMockLogsExporter(t *testing.T, cfg Config, host component.Host, topics ...string) (*kafkaExporter[plog.Logs], *kfake.Cluster) {
+	exp := newLogsExporter(cfg, exportertest.NewNopSettings(metadata.Type))
+	cluster := configureExporter(t, exp, cfg, host, topics...)
+	return exp, cluster
+}
 
-		got := wrapKafkaProducerError(prodErrs)
+func newKgoMockTracesExporter(t *testing.T, cfg Config, host component.Host, topics ...string) (*kafkaExporter[ptrace.Traces], *kfake.Cluster) {
+	exp := newTracesExporter(cfg, exportertest.NewNopSettings(metadata.Type))
+	cluster := configureExporter(t, exp, cfg, host, topics...)
+	return exp, cluster
+}
 
-		assert.False(t, consumererror.IsPermanent(got))
-		assert.Contains(t, got.Error(), err.Error())
-	})
+func newKgoMockMetricsExporter(t *testing.T, cfg Config, host component.Host, topics ...string) (*kafkaExporter[pmetric.Metrics], *kfake.Cluster) {
+	exp := newMetricsExporter(cfg, exportertest.NewNopSettings(metadata.Type))
+	cluster := configureExporter(t, exp, cfg, host, topics...)
+	return exp, cluster
+}
 
-	t.Run("should return not permanent error on other producer error", func(t *testing.T) {
-		err := errors.New("other producer error")
-		prodErrs := sarama.ProducerErrors{
-			&sarama.ProducerError{Err: err},
-		}
+func configureExporter[T any](tb testing.TB,
+	exp *kafkaExporter[T], cfg Config, host component.Host, topics ...string,
+) *kfake.Cluster {
+	cluster, kcfg := kafkatest.NewCluster(tb, kfake.SeedTopics(1, topics...))
 
-		got := wrapKafkaProducerError(prodErrs)
+	// Create a kgo.Client using the broker addresses from the fake cluster.
+	kgoClientOpts := []kgo.Opt{
+		kgo.SeedBrokers(kcfg.Brokers...),
+		kgo.ClientID(cfg.ClientID),
+	}
+	client, err := kgo.NewClient(kgoClientOpts...)
+	require.NoError(tb, err, "failed to create kgo.Client with fake cluster addresses")
 
-		assert.False(t, consumererror.IsPermanent(got))
-		assert.Contains(t, got.Error(), err.Error())
-	})
+	messager, err := exp.newMessager(host) // messager implements Marshaler[pmetric.Metrics]
+	require.NoError(tb, err, "failed to create messager for metrics")
 
-	t.Run("should return not permanent error when other error", func(t *testing.T) {
-		err := errors.New("other error")
+	exp.messenger = messager
+	exp.producer = kafkaclient.NewFranzSyncProducer(client, cfg.IncludeMetadataKeys)
 
-		got := wrapKafkaProducerError(err)
+	tb.Cleanup(func() { assert.NoError(tb, exp.Close(context.Background())) })
+	return cluster
+}
 
-		assert.False(t, consumererror.IsPermanent(got))
-		assert.Contains(t, got.Error(), err.Error())
+// fetchKgoRecords polls a franz-go topic for up to 5 seconds and returns all records produced to that topic.
+func fetchKgoRecords(tb testing.TB, brokers []string,
+	topic, group string, opts ...kgo.Opt,
+) []*kgo.Record {
+	clientOpts := []kgo.Opt{
+		kgo.SeedBrokers(brokers...),
+		kgo.ConsumeTopics(topic),
+		kgo.ConsumerGroup(group),
+		kgo.FetchMaxBytes(10 * 1024 * 1024),
+	}
```

Comment:
Could we simplify callers by using `t.Name()` for the consumer group name here, replacing restricted characters as needed?

Reply:
Actually, do we even need to do that much? Maybe just use a hardcoded group name here, and tests can override that if needed?